src/data/data_representation_converter.cpp

```diff
@@ -0,0 +1,436 @@
+/*
+ * Copyright 2025, Sirius Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "data/data_representation_converter.hpp"
+#include "memory/result_table.hpp"
+#include "memory/fixed_size_host_memory_resource.hpp"
+
+#include "duckdb/common/types/string_type.hpp"
+#include "duckdb/common/types.hpp"
+
+#include <cuda_runtime.h>
+#include <cudf/column/column_view.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/bit.hpp>
+
+#include <cmath>
+#include <unordered_map>
+
+namespace sirius {
+
+using sirius::memory::fixed_size_host_memory_resource;
+
+constexpr size_t BYTE_TO_BITS = 8;
+
+// Helper map to map CUDF types to DuckDB types. TODO(Devesh): Make sure that all of the mapping make sense because not all of them are 1:1 mappings
+const std::unordered_map<cudf::type_id, duckdb::LogicalTypeId> CUDF_TO_DUCKDB_TYPES_MAP = {
```

Comment:
This mapping should be handled somewhere else and not in the duckdb data representation. This understanding of how one type systems maps to another should be handled in one place so that when it needs to expand to include some other type system that becomes easier to do so.  Some of these mappings don't have valid conversions and so we need to be able to decide how we will deal with types that don't have conversions

Reply:
I do agree that ideally we need some sort of `type_converter` and associated `type_converter_registry` similar to what you have started implementing in https://github.com/NVIDIA/cuCascade/pull/4. 

However, what the interface for what is unclear to me because currently we don't have a concrete representation of consists of a type in Sirius? I think we do need to discuss how we represent type systems and types in a way universal way so that we can support for Spark's type system in the future but I am not sure if it makes sense in this PR or in another PR